rewrite-java/src/main/java/org/openrewrite/java/CreateJavaClass.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.intellij.lang.annotations.Language;
+import org.openrewrite.CreateFileVisitor;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.ScanningRecipe;
+import org.openrewrite.SourceFile;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.StringUtils;
+import org.openrewrite.internal.lang.NonNull;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.tree.J;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Collections.emptyList;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class CreateJavaClass extends ScanningRecipe<AtomicBoolean> {
```

Comment:
This is missing a way for the parser classpath to be specified, which will be necessary for most non-trivial java classes.

Reply:
How that can be done? I tried an implementation here, but the scanning recipe was passing along `SourceSpec`s with parse errors.

I also tried to enable logging for compilation errors (the parser supports it), but 1) there is no logging implementation on `rewrite-java`; and 2) I tried adding `testRuntimeOnly("org.slf4j:slf4j-simple:1.7+")` on `rewrite-java`'s `build.gradle.kts`, and it also didn't work, because archunit is [pulling slf4j-api 2.0.3](https://central.sonatype.com/artifact/com.tngtech.archunit/archunit/1.0.1/dependencies)and SLF4J 1.x bindings don't work on SLF4J 2.x.  Any advice?

Another question, how do I enable `TypeValidation` on the tests? Adding `spec.typeValidationOptions(TypeValidation.builder().build())` didn't work.